Sources/Sentry/SentryTracer.m

```diff
@@ -281,6 +283,9 @@ - (void)spanFinished:(id<SentrySpan>)finishedSpan
     // Calling canBeFinished on the rootSpan would end up in an endless loop because canBeFinished
     // calls finish on the rootSpan.
     if (finishedSpan != self.rootSpan) {
+        // If this is the last child span to be finished, cache its status in case this trace does
+        // not get finish with a status called directly
+        _finishStatus = finishedSpan.context.status;
```

Comment:
h: This is not align with specs. Each span has it status and the transaction itself has it own status.
Also, this will override the rootSpan status in case we are using `waitForChildren`.

```suggestion
```

Reply:
That is a good point about the `rootSpan` override. I had missed that. Will revert.